src/native/libs/System.Native/CMakeLists.txt

```diff
@@ -57,6 +57,8 @@ else()
         pal_signal_wasi.c
         pal_threading_wasi.c
         pal_uid_wasi.c
+        pal_cabi_wasi.h
+        pal_cabi_wasi.c
```

Comment:
I made `cabi_realloc` part of the PAL. 

@SingleAccretion is that OK for now ?

Reply:
The location is good, but I suspect it doesn't work as-is. `libSystem.Native.a` is an archive, which means the linker won't look at the object files produced for `pal_cabi_wasi.c` unless something in the static link pulls in `cabi_realloc`.

Possible workaround:
1) Add a new object (not static) file library just for this symbol so that it is always linked in.
2) Add `--undefined=cabi_realloc` to the final linker command line.

Reply:
added `--undefined=cabi_realloc` 
How can I check it helped ?

Reply:
It shouldn't have worked without it. The expected state is, of course, that the component exports `cabi_realloc`.

Reply:
I see `(export "cabi_realloc" (func $cabi_realloc))` in the core module now, so that's fine.

But the sample was able to run even before I added the linker flag.
It is in-tree sample and it doesn't re-link the dotnet.wasm for the application, so maybe that's why ?

Reply:
Checking what pulls in the symbol (without `--undefined`) with `--trace-symbol=cabi_realloc` would tell us why it worked.

Reply:
It said
```
libSystem.Native.a(pal_cabi_wasi.c.obj): lazy definition of cabi_realloc
```

maybe `__attribute__((__weak__, __export_name__("cabi_realloc")))` does some magic ?

Reply:
I guess there is also `--why-extract=<log file path>`, but it's fine to leave as-is pending actual problems.

> maybe __attribute__((__weak__, __export_name__("cabi_realloc"))) does some magic ?

I wouldn't expect it to.

Reply:
got `artifacts\bin\native\net9.0-wasi-Debug-wasm\extract.txt` but it doesn't mention `cabi_realloc` at all. It does mention `libSystem.Native.a` in both imports/exports directions.

So I guess we are getting some other `cabi_realloc`. I will try to damage it and see if it still "works".

Reply:
`C:/Dev/runtime/src/mono/wasi/wasi-sdk/share/wasi-sysroot/lib/wasm32-wasip2/libc.a(wasip2.o): definition of cabi_realloc` I missed that message before

Reply:
So it is defined in libc (as was suspected), here: https://github.com/WebAssembly/wasi-libc/blob/320bbbcced68ce8e564b0dc4c8f80a5a5ad21a9c/libc-bottom-half/sources/wasip2.c#L372-L378. I guess we don't really need this definition then.

Reply:
Ah, good find.  I put that file there, so I should have known 🤦

Reply:
I added dummy usage of `cabi_realloc` in `src/mono/wasi/runtime/driver.c` and that fixed it.

Reply:
ok, I will remove it from the PR then

Reply:
> I added dummy usage of cabi_realloc in src/mono/wasi/runtime/driver.c and that fixed it.

That's another way to have this be more robust. `pal_cabi_wasi.c` can be deleted now, though.